swingset/src/main/java/com/nqadmin/swingset/utils/SSCommon.java

```diff
@@ -395,31 +419,64 @@ public SSCommon(final SSComponentInterface _ssComponent) {
 //	}
 
 	/**
-	 * Methold to add a document listener when the SwingSet component is a
+	 * Method to add a document listener when the SwingSet component is a
 	 * JTextComponent.
 	 */
-	public void addDocumentListener() {
-		((javax.swing.text.JTextComponent) getSSComponent()).getDocument().addDocumentListener(documentListener);
+	private void addSSDocumentListener() {
+		((javax.swing.text.JTextComponent) getSSComponent()).getDocument().addDocumentListener(ssDocumentListener);
 	}
 
 	/**
 	 * Method to add the RowSet listener.
 	 */
 	public void addRowSetListener() {
-		if (!rowsetListenerAdded) {
+		if (rowSetListener==null) {
+			rowSetListener = new SSRowSetListener();
+		}
+		if (!rowSetListenerAdded && rowSet!=null) {
 			rowSet.addRowSetListener(rowSetListener);
-			rowsetListenerAdded = true;
+			rowSetListenerAdded = true;
+			logger.debug(boundColumnName + " - RowSet Listener added.");
 		}
 	}
 	
 	/**
 	 * Method to add any SwingSet Component listener(s).
 	 */
-	public void addSSComponentListener() {
+	public final void addSSComponentListener() {
```

Comment:
**So if you study one thing carefully in this PR, this method should be that thing. And the removeSSComponentListener() counterpart.**